Доработана функциональность по ТЗ спринта 7

iGladkikh opened the pull request.

iGladkikh committed:
Доработать функциональность по ТЗ спринта 7

Yang-Pi commented on src/services/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,263 @@
+        throw new ManagerLoadException("Файл '" + file + "' отсутствует.");
+    }
+
+    public static String historyToString(HistoryManager manager) {
```
Слишком широкий модификатор доступа у метода (и у двух ниже)

iGladkikh replied:
Удален модификатор "public". Спасибо!

Yang-Pi commented on src/services/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,263 @@
+
+                    if (epicType.equals(startWorld)) {
+                        Epic epic = Epic.createFromString(taskData, DATA_FILE_DELIMITER);
+                        addEpic(epic);
```
Обрати внимание, что внутри методов add*() вызывается логика сохранения, те при загрузки задач из файла задачи в этот же файл одновременно и сохраняются.

iGladkikh replied:
Исправлено на вызов методов "super.add*()". Спасибо!

Yang-Pi commented on src/services/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,263 @@
+                        addSubtask(subtask);
+                        loadedTasks.put(subtask.getId(), subtask);
+                    } else if (taskType.equals(startWorld)) {
+                        Task task = Task.createFromString(taskData, DATA_FILE_DELIMITER);
```
Важный момент - для задач был задан id, но затем при создании новых будет id cнова генерироваться, но при текущей реализации эта генерация будет с нуля, что приведет к дублированию.

iGladkikh replied:
Внесены дополнения в логику конструктора класса Task. Спасибо!

iGladkikh committed:
Внести исправления после ревью 7.1:
1. Ограничить уровень доступа к статическим методам класса FileBackedTaskManager.
2. Исправить логику восстановления данных из файла (вызывать методы "super.add*()" вместо "add*())".
3. Исправить порядок генерации id последующих задач при создании задачи из строки.

iGladkikh committed:
Добавить автотесты.
